src/hooks/use-water-block.hook.ts

```diff
@@ -0,0 +1,150 @@
+import { useCallback, useState } from 'react'
+import { useAppContext } from '../context'
+import dayjs from 'dayjs'
+import { v4 as uuidv4 } from 'uuid'
+import { TWater } from '../Types/DataUserTypes'
+import { DataType, OperationType } from '../shared/enums'
+import { useDispatch, useSelector } from 'react-redux'
+import { operationData } from '../store/slices/dataUserSlice'
+import { getUserWater } from '../store/selectors/selectors'
+import { addDoc, collection, doc, getDocs, query, updateDoc, where } from 'firebase/firestore'
+import { db } from '../firebase/firebaseConfig'
+
+type WaterBlock = {
+  waterMl: number | undefined
+  allWater: Array<TWater>
+  handleChangeWaterInput: (event: React.ChangeEvent<HTMLInputElement>) => void
+  todayHandleClick: VoidFunction
+  clearButtonHandler: VoidFunction
+  addButtonHandler: VoidFunction
+  deleteHandler: (id: string) => void
+  getUserWaterDB: (date: string) => Promise<void>
+}
+
+function useWaterBlock(): WaterBlock {
+  const [waterMl, setWaterMl] = useState<number | undefined>()
+  const { selectedDate, handleDateChange } = useAppContext()
+  const dispatch = useDispatch()
+  const allWater = useSelector(getUserWater)
+  const email = localStorage.getItem('email')
+
+  const handleChangeWaterInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = Number(event.target.value)
+    if (inputValue >= 0) {
+      setWaterMl(inputValue)
+    }
+  }
+  const todayHandleClick = () => {
+    handleDateChange(dayjs())
+  }
+
+  const clearButtonHandler = () => {
+    setWaterMl(undefined)
```

Comment:
we can't set to `undefined`, lets check different between `null` and `undefined`